saml/src/main/java/com/linecorp/armeria/saml/HttpPostBindingUtil.java

```diff
@@ -0,0 +1,142 @@
+import org.opensaml.saml.common.SAMLObject;
+import org.opensaml.saml.common.SignableSAMLObject;
+import org.opensaml.saml.common.messaging.context.SAMLBindingContext;
+import org.opensaml.security.credential.Credential;
+
+import net.shibboleth.utilities.java.support.codec.Base64Support;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.util.Exceptions;
+
+/**
+ * A utility class which supports HTTP-Post binding protocol.
+ */
+final class HttpPostBindingUtil {
+
+    private static final String[] HTML = {
+            // 0
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
+            "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\" \"DTD/xhtml1-strict.dtd\">" +
+            "<html xmlns=\"http://www.w3.org/1999/xhtml\" xml:lang=\"en\" lang=\"en\">" +
+            "<head><meta http-equiv=\"content-type\" content=\"text/html;charset=utf-8\" /></head>" +
+            "<body onload=\"document.forms[0].submit()\">",
+            // 1, 2
+            "<form method=\"post\" action=\"", "\">",
+            // 3, 4, 5
+            "<input type=\"hidden\" name=\"", "\" value=\"", "\" />",
+            // 6
+            "</form></body></html>"
+    };
+
+    /**
+     * Returns an {@link HttpData} which holds an HTML form content.
+     */
+    static HttpData getHtmlForm(String remoteEndpointUrl,
+                                String paramName, String paramValue,
+                                @Nullable String relayState) {
+        requireNonNull(remoteEndpointUrl, "remoteEndpointUrl");
+        requireNonNull(paramName, "paramName");
+        requireNonNull(paramValue, "paramValue");
+
+        final StringBuilder html = new StringBuilder();
+        html.append(HTML[0])
+            .append(HTML[1]).append(remoteEndpointUrl).append(HTML[2])
+            .append(HTML[3]).append(paramName)
+            .append(HTML[4]).append(paramValue).append(HTML[5]);
+
+        if (relayState != null) {
+            html.append(HTML[3]).append(PARAM_RELAY_STATE)
+                .append(HTML[4]).append(relayState).append(HTML[5]);
+        }
+        html.append(HTML[6]);
+
+        return HttpData.ofUtf8(html.toString());
+    }
+
+    /**
+     * Signs the specified {@link SignableSAMLObject} with the specified {@link Credential} and
+     * {@code signatureAlgorithm}, and then encode the object as base64 format.
+     */
+    static String signAndBase64Encode(SignableSAMLObject signableObj,
+                                      Credential signingCredential,
+                                      String signatureAlgorithm) {
+        sign(signableObj, signingCredential, signatureAlgorithm);
+        try {
+            final String messageStr = nodeToString(XMLObjectSupport.marshall(signableObj));
+            return Base64Support.encode(messageStr.getBytes(), Base64Support.UNCHUNKED);
+        } catch (Exception e) {
```

Comment:
Global comment, I think the code would be easier to reason about if we use multicatch instead of `Exception/Throwable` throughout.